File: static/plugins/grid/grid.js
// grid.js
// by Ben Crowder
 
$(document).ready(function() {
	$(".entry .content").each(function() {
		grid_init($(this));
	});
});

function grid_init(content) {
    /*
	content.find(".grid-box").each(function() {
		// Grab the content
		var content = $.trim($(this).html());
		var lines = content.split('\n');
		
		// Parse
		html = '<table>';
		for (var i in lines) {
			var line = $.trim(lines[i]);
	
			html += '<tr>';
			for (var x in line) {
				html += '<td';
				if (line[x] == '1') {
					html += ' class="filled"';
				}
				html += '></td>';
			}
			html += '</tr>';
		}
		html += '</table>';

		// replace the div's contents with the generated HTML
		$(this).html(html);
	});
    */
}

File: plugins/grid.py
# Grid plugin

import re

# Convert the 010100 etc. syntax to an HTML table
def gridify(m):
    content = m.group(1)
    lines = content.split('\n')

    html = '<div class="grid-box">'
    html += '<table>'

    for line in lines:
        html += '<tr>'
        for char in line:
            html += '<td%s></td>' % (' class="filled"' if char == '1' else '')
        html += '</tr>'
    html += '</table>'
    html += '</div>'

    return html

def process(content, entry, notebook_url):
    # Convert {grid ... } notation to the HTML we need
    regex = re.compile(r'{grid\n(.*?)\n}', flags=re.DOTALL)
    content = regex.sub(gridify, content)

    return content
